hoomd/hpmc/ExternalFieldJIT.h

```diff
@@ -161,7 +163,12 @@ template<class Shape> class ExternalFieldJIT : public hpmc::ExternalFieldMono<Sh
             // read in the current position and orientation
             Scalar4 postype_i = h_postype.data[i];
             unsigned int typ_i = __scalar_as_int(postype_i.w);
-            vec3<Scalar> pos_i = vec3<Scalar>(postype_i);
+            int3 image = make_int3(0, 0, 0);
+            vec3<Scalar> pos_i = vec3<Scalar>(postype_i) - vec3<Scalar>(this->m_pdata->getOrigin());
+            box_new.wrap(pos_i, image);
+            image = make_int3(0, 0, 0);
+            vec3<Scalar> old_pos_i = vec3<Scalar>(*(position_old + i));
```

Comment:
* `old_pos_i` must be shifted bo the old origin, which the caller must provide.
* `IntegratorHPMC::attemptBoxResize` should scale the origin.